File: src/main/java/land/leets/domain/auth/UserAuthDetailsService.java
package land.leets.domain.auth;

import land.leets.domain.shared.AuthRole;
import land.leets.domain.user.domain.User;
import land.leets.domain.user.domain.repository.UserRepository;
import land.leets.global.error.ErrorCode;
import lombok.RequiredArgsConstructor;
import org.springframework.security.core.userdetails.UserDetailsService;
import org.springframework.security.core.userdetails.UsernameNotFoundException;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class UserAuthDetailsService implements UserDetailsService {
    private final UserRepository userRepository;

    @Override
    public AuthDetails loadUserByUsername(String email) throws UsernameNotFoundException {
        User user = this.userRepository
                .findByEmail(email)
                .orElseThrow(() -> new UsernameNotFoundException(ErrorCode.USER_NOT_FOUND.getMessage()));
        return new AuthDetails(user.getUid(), user.getEmail(), AuthRole.ROLE_USER);
    }
}

File: src/main/kotlin/land/leets/domain/auth/UserAuthDetailsService.kt
package land.leets.domain.auth

import land.leets.domain.shared.AuthRole
import land.leets.domain.user.domain.repository.UserRepository
import land.leets.global.error.ErrorCode
import org.springframework.security.core.userdetails.UserDetailsService
import org.springframework.security.core.userdetails.UsernameNotFoundException
import org.springframework.stereotype.Service

@Service
class UserAuthDetailsService(
    private val userRepository: UserRepository
) : UserDetailsService {

    override fun loadUserByUsername(email: String): AuthDetails {
        val user = userRepository.findByEmail(email)
            ?: throw UsernameNotFoundException(ErrorCode.USER_NOT_FOUND.message)

        return AuthDetails(user.id!!, user.email, AuthRole.ROLE_USER)
    }
}
